### PR TITLE
(Frontend) CAWP

### LINKED ISSUE #1441: Epic: Women Legislators Represenation
## Background

[View Google Doc](https://docs.google.com/document/d/1W3qqsOnZobX0ZI09XPNTI-s8AOoHRTjl/edit?usp=sharing&ouid=108877694621640419570&rtpof=true&sd=true)

#1512 
#1518 

## Proposed Solution

Incorporating Aye and team's findings into the tracker, starting with the prevalence of women legislators. 

## End State 

User can select `% Women Legislators` from the dropdown menu and view reports at the National and State level that include maps, charts and data tables representing the secondary data Aye' sourced

## How will this move the needle on health equity?
Specifically how this will help advance health equity, and is possible who the target audience might be (policy influencers, non-technical users, etc)

## Source(s)
- (priority) State Legislatures: https://cawp.rutgers.edu/facts/levels-office/state-legislature/women-state-legislatures-2022#table
- (future) US Congress: https://cawp.rutgers.edu/facts/levels-office/congress/women-serving-117th-congress-2021-2023

### PR DESCRIPTION
**Pull request by benhammondmusic**
## Preview Link

<a href="https://deploy-preview-1518--health-equity-tracker.netlify.app/exploredata?mls=1.women_in_legislative_office-3.00&dt1=women_state_legislatures" target="_blank">
  <br />
Click to demo updates </a> 

## Frontend Tasks:
- [x] bug with doubling Michigan because CAWP used `MI` for both Michigan and `Mariana Islands - MI` 
- [x] Add `% Women Legislators` as new topic under the `Political Determinants of Health` Category
- [x] Card rendering logic; which cards appear at which levels? Hidden entirely if no data, or with no data alert message as normal? 
- [x] National report shows a national map by state (clickable); state report shows a state map by county (not-clickable); 
- [x] bar chart by race (by age?) showing total number of women leg. for each race rather than a `per 100k` rate. 
- [x] no information to display for unknowns map? or could we map the 18 unknowns somehow
- [x] disparity bar chart could be interesting, showing a race's `% of total population` vs `% of female legislators` 
- [x] Confirm new, alternate race labels are working
- [x] confirm flow: remove/disable demographic toggle. Only relevant option is for `race` unless we can reliably source each legislators birthday (seems tedious and not all that helpful?)
- [x] Confirm: to start we will NOT use data type toggle and will only do STATE legislatures. eventually we might follow up by adding new data types For `US Congress`, possibly `Mayors`, `elected positions` etc as options
- [x] manipulate the data table card to display different information (not including the % share metrics and instead visualizing each of the types of legislators on a single table `US Congress`, `State Legislature`, `Mayor`)
- [x] create a frontend Provider file that will receive data from DataFetcher and route it properly
- [x] edit DataFetcher to handle typing for this new data coming from BigQuery
- [x] will we require frontend tests (for any of the topics) if all of our calculations are done on the backend? 
- [x] #1529 ?
- [x] finalize all titles / labels (discussion / decision in [this Google Doc](https://docs.google.com/document/d/1rt1uofLiJIya_4vqO2FI0mEggkSQvtZehzRWJgIF-_k/edit?usp=sharing))
- [x] consider a third bar on the comparison bar chart to show population of {race women} / {women}
- [x] consider how to fill in the missing populations for the CAWP non-standard race groups
- [x] consider adding `women` to each race group label for clarity; issue with the population bar though
- [x] adjust "over 100%" alert to address that every race can be double counted, not only hispanic
- [ ] troubleshoot map legend not scaling properly
- [x] make compare mode bar chart use women-only labels
- [x] sometimes getting issue where vega tooltip only flashes instead of persisting. seemed to be once i have scrolled down the page and back up, and seemed related to the computer working extra hard. maybe has gone away since i restarted my machine so possible it's only in dev and not in the real build
- [x] figure out what to do with [states that have no women](https://deploy-preview-1518--health-equity-tracker.netlify.app/exploredata?mls=1.women_in_legislative_office-3.16&dt1=women_us_congress) in that legislature

## Description

Frontend for new data from Center for American Women in Politics

## Motivation and Context

closes #1441 

## How has this been tested?
- frontend tests pass; added tests for any added frontend functions
- manually on MacOS Chrome, and simulated mobile

## Types of changes

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by benhammondmusic** [comment]
Screen Shot 2022-03-15 at 7 47 36 PM

**Comment by benhammondmusic** [comment]
Screen Shot 2022-03-15 at 7 57 49 PM

**Comment by benhammondmusic** [comment]
Screen Shot 2022-03-15 at 10 30 50 PM

**Comment by benhammondmusic** [comment]
Using ACS numbers on the backend to attach the pct_population properties for each race/state. It can only do standard census races, so we need to brainstorm how to (or if we even can) present population metrics for these non-standard races.

In another sense, we might need to consider what the comparison population should even be:
- all people in a geo
- All women in a geo
- All legislators in a geo
- All women legislators in a geo

From slack: 
Ben: "there are arguments either way. I think our current pattern would be to use women leg as the denominator; but this is the first time we've shown intersectionality and the “goal” of this is to show disparity of women in gov , amplified by race (I think) 

Which kind of begs the question: why aren't we just doing demographic makeup of legislators, which we could then filter by age by sex and by race ?

I think because non-white women are most under represented

But if we do go this way we’ll have to be really clear with alerts etc because you're right it's a different way of doing it"

Screen Shot 2022-03-16 at 4 40 43 PM

**Comment by benhammondmusic** [comment]
National Population Share working

Screen Shot 2022-03-16 at 9 32 01 PM

**Comment by benhammondmusic** [comment]
Updates after deciding on our metrics. Wording / labels still needs a lot of tweaking:
Screen Shot 2022-03-22 at 4 45 59 PM
Screen Shot 2022-03-22 at 4 47 11 PM

**Comment by benhammondmusic** [comment]
Got a second data type working: `US Congress`

Screen Shot 2022-03-23 at 8 41 31 PM
Screen Shot 2022-03-23 at 8 41 51 PM
Screen Shot 2022-03-23 at 8 42 13 PM
Screen Shot 2022-03-23 at 8 42 26 PM
Screen Shot 2022-03-23 at 8 42 33 PM

**Comment by benhammondmusic** [comment]
Adjusting labels:
- remove term "congresswomen" as it's not a universally preferred term
- help differentiate gov levels by using `US Congress` vs `State Legislator / State Legislature`
- screenshots of what adding women-specific race groups would look like

Screen Shot 2022-03-24 at 2 04 49 PM
Screen Shot 2022-03-24 at 2 04 59 PM
Screen Shot 2022-03-24 at 2 05 11 PM
Screen Shot 2022-03-24 at 2 05 21 PM

**Comment by benhammondmusic** [comment]
Initial Spot Checks; I've been including territories in both the numerator and denominator but CAWP does not which likely account for the small discrepancies at the national level. We need to decide if we should include these territories in the national rates (in US Congress they are non-voting but can participate in committees and represent their constituents at the federal level)

Screen Shot 2022-03-25 at 11 06 07 PM
Screen Shot 2022-03-25 at 11 08 13 PM
Screen Shot 2022-03-25 at 11 10 28 PM

**Comment by benhammondmusic** [comment]
Updated data eliminating double counts, edge cases, and refactoring for easier to follow calculations

Screen Shot 2022-03-25 at 11 12 50 PM
Screen Shot 2022-03-25 at 11 12 54 PM
Screen Shot 2022-03-25 at 11 12 59 PM

Screen Shot 2022-03-25 at 11 12 46 PM

Screen Shot 2022-03-25 at 11 12 42 PM
; 
Screen Shot 2022-03-25 at 11 12 15 PM
Screen Shot 2022-03-25 at 11 12 29 PM

Screen Shot 2022-03-25 at 11 12 04 PM
Screen Shot 2022-03-25 at 11 12 11 PM

**Comment by benhammondmusic** [comment]
Discussion and collaboration on labeling and terminology in [this Google Doc](https://docs.google.com/document/d/1rt1uofLiJIya_4vqO2FI0mEggkSQvtZehzRWJgIF-_k/edit?usp=sharing)

**Comment by benhammondmusic** [comment]
## Updates:
- uses women-specific race labels on bar chart
- identifies selected race on Map Card
- adjusts labels to match current best choices

Screen Shot 2022-04-07 at 2 16 28 PM

Screen Shot 2022-04-07 at 2 16 12 PM

Screen Shot 2022-04-07 at 2 16 03 PM


Screen Shot 2022-04-07 at 2 15 39 PM


Screen Shot 2022-04-07 at 2 15 01 PM
Screen Shot 2022-04-07 at 2 14 47 PM

**Comment by benhammondmusic** [comment]
> Women only race labels on the MultiMap modal view:
Screen Shot 2022-04-13 at 9 01 17 AM

**Comment by benhammondmusic** [comment]
Screen Shot 2022-04-20 at 12 18 08 PM

**Comment by benhammondmusic** [comment]
First draft for methodology. Would like to work with Aye to include a "Why?" section re: the importance of measuring this metric in terms of its legislative impact
Screen Shot 2022-04-20 at 1 13 36 PM

Blurb at the bottom of the tracker in the "missing data" section
Screen Shot 2022-04-20 at 12 18 08 PM

**Comment by benhammondmusic** [comment]
Both data types now better handle the "race only" breakdown limitation:
- the AgeAdjusted card elegantly shows only the Unknown Race without mentioning the non-existent Unknown Age
- the toggle select disables `Age` and `Sex`. I had hidden them entirely before, but the issue with that is a user can selected `Age` on a different health topic like `Excessive Drinking`, and then select `Women in Legislative Office` and the `Age` filter stays in place. Then, because the toggle is hidden, the user has no way of changing to `Race`. By disabling the non-options, the user can still select `Race` if they accidentally end up on `Age` or `Sex`

Screen Shot 2022-04-21 at 11 27 43 AM

**Comment by benhammondmusic** [comment]
> @benhammondmusic I'm not sure about this visualization for states with zero women legislators 
> 
> 
> 
> Screen Shot 2022-04-28 at 3 35 37 PM
> 
> 

image

File: frontend/.env.deploy_preview [hunk 1]
```diff
@@ -2,4 +2,6 @@ REACT_APP_DEPLOY_CONTEXT=deploy_preview
 REACT_APP_BASE_API_URL=https://dev.healthequitytracker.org
```
[the remark this answers is not in the document]
Comment (benhammondmusic) [reply frontend/.env.deploy_preview]: I dont think I cahnged this...

[the remark this answers is not in the document]
Comment (joshzarrabi) [reply frontend/.env.deploy_preview]: Thats old, sorry i forget that the commends don't get published until you submit the review

**Pull request merged**